### ISSUE TITLE
feat: animate `box-shadow`

### ISSUE THREAD

**Issue by std-microblock** [mikke89/RmlUi#688]
currently the `box-shadow` property is not animatable.

```log
[I][2024-10-04 23:36:26.6798450] Element animation was not initialized properly, can't add key.
[I][2024-10-04 23:36:27.7997935] Property value '' is not a valid target for interpolation.
[I][2024-10-04 23:36:27.8000657] Element animation was not initialized properly, can't add key.
[I][2024-10-04 23:36:28.4354317] Property value '#ffffff79 0dp 0dp 25dp 3dp' is not a valid target for interpolation.
[I][2024-10-04 23:36:28.4359164] Element animation was not initialized properly, can't add key.
```

**Comment by std-microblock** [comment]
> Right, this is a known and documented limitation currently: https://mikke89.github.io/RmlUiDoc/pages/rcss/animations_transitions_transforms.html#animation
> 
> I have some concerns about straight-forwardly making it animatable. Particularly in terms of performance, since this would create a new render texture every frame. Maybe as an optimization we could render it live, without rendering to texture first, whenever it is animated. Or maybe it doesn't actually matter in terms of performance, and it's all fine.

I wondered how did real-world browsers like chromium or firefox implemented this? We should take a look into that.
In my another project this was implemented by generate a shader with a input variable _progress_. Can we implement this the same way? Idk what rml is doing right now.

**Comment by std-microblock** [comment]
Also if this is not supported yet, we should simply ignore it instead of logging a confusing error every time it changes..

**Comment by mikke89** [comment]
Hey, you're right that the warning message was a bit cryptic. So I went ahead and improved that.

As for the implementation details, it's not really a question of how to do it. We know that already, it's just about the effort and measuring the performance impact, and see if we need to handle animations specially when generating the box shadows.

With that, I am closing this one. The confusing error message has been addressed, and we don't need to keep issues open for known and documented features that have not yet been implemented.
